crates/gpui/src/elements/text.rs

```diff
@@ -305,11 +305,17 @@ impl TextLayout {
                 }
 
                 let mut line_wrapper = cx.text_system().line_wrapper(text_style.font(), font_size);
-                let text = if let Some(truncate_width) = truncate_width {
-                    line_wrapper.truncate_line(text.clone(), truncate_width, ellipsis)
+                let (text, truncated) = if let Some(truncate_width) = truncate_width {
+                    (
+                        line_wrapper.truncate_line(text.clone(), truncate_width, ellipsis),
+                        true,
+                    )
                 } else {
-                    text.clone()
+                    (text.clone(), false)
                 };
+                if truncated {
+                    runs[0].len = text.len();
```

Comment:
Is this safe?

Reply:
This is definitely wrong, should properly handle the `runs` here, need more investigating...